ERCS/erc-7531.md

```diff
@@ -0,0 +1,124 @@
+  // previous HolderOfRightsFor event. 
+  // Since any smart contract could emit a fake HolderOfRightsFor event, entities listening to the event, 
+  // MUST check that the contract emitting the event is consistent with the current owner of the token coming
+  // from the related Transfer event.
+  event HolderOfRightsFor(address indexed tokenAddress, uint256 indexed tokenId, address indexed holder);
+  
+  // @dev Returns the address of the entity with rights over the token, when it's not the current owner.
+  //   It MUST revert if the token is not held.
+  // @param tokenAddress The address of the ERC-721 contract.
+  // @param tokenId The ID of the token.
+  // @return The address of the entity with rights over the token.
+  function holderOfRightsFor(
+    address tokenAddress,
+    uint256 tokenId
+  ) external view returns (address);
+}
+```
+
+The `HolderOfRightsFor` event is crucial for accurately identifying the actual owner of a held token. In scenarios where a token is staked in a contract, the [ERC-721](./eip-721.md) Transfer event would incorrectly assign ownership to the staking contract itself. The `HolderOfRightsFor` event addresses this discrepancy by explicitly signaling the real owner of the token rights.
+
+### Timing of Event Emission:
+
+The `HolderOfRightsFor` event MUST be emitted either in the same block as the corresponding Transfer event or in any subsequent block. This approach offers flexibility for existing pools to upgrade their systems without compromising past compatibility. Specifically, staking pools can emit this event for all previously staked tokens, or they can allow users to actively reclaim their ownership. In the latter case, the event SHOULD be emitted as part of the ownership reclamation process. This flexibility ensures that the system can adapt to both current and future states while accurately reflecting the actual ownership of held tokens.
+
+### Invalidation of Previous HolderOfRightsFor Events:
+
+To maintain compatibility with the broader ecosystem and optimize for gas efficiency, any new `Transfer` event involving the same token invalidates the previous `HolderOfRightsFor` event. This approach ensures that the most recent `Transfer` event reliably reflects the current ownership status, negating the need for additional events upon unstaking.
+
+### Important Consideration Regarding Event Authenticity:
+
+Given that any smart contract has the ability to emit a `HolderOfRightsFor` event, it is crucial for entities relying on these events to validate their authenticity. Specifically, entities should verify that the contract emitting a `HolderOfRightsFor` event is indeed the current holder (i.e., the technical owner) of the token in question, for example, comparing the emitting contract with the owner resulting from the most recent `Transfer` event. This verification step is essential to ensure that the information provided by the event is accurate and not fabricated by an unauthorized source.
```

Comment:
Does this belong in the security considerations section?

Reply:
Good point. I will move it.